papermillhub/app.py

```diff
@@ -73,6 +73,15 @@ def _normalize_base_url(self, proposal):
         url = urlunparse(parsed._replace(path=parsed.path.rstrip("/")))
         return url
 
+    # TODO: figure out how to make this unnecessary. In a perfect world,
```

Comment:
This is really unfortunate. We need access to the JupyterHub public proxy API in order to make requests of the user servers (this is the same issue as https://github.com/jupyterhub/jupyterhub/issues/2611). I couldn't figure out a way around this, but if you have any ideas, I'm all ears.